demo/react-hooks-controlled-multiple.html

```diff
@@ -68,45 +69,35 @@ <h2>Controlled stack</h2>
           });
         })
         .on('removed', (ev, gsItems) => {
+          /* Looks like a bug in GridStack */
+          const dirtyNodes = grid.engine.getDirtyNodes();
+          if (dirtyNodes !== undefined && dirtyNodes.length !== 0) {
+            const newItems = grid.save(false);
+            changeItems(newItems);
+          }
+
           if (grid._ignoreCB) return;
           gsItems.forEach(n => removeItem(n.id));
         })
-
+        .on('change', (ev, gsItems) => {
+          const newItems = grid.save(false);
+          changeItems(newItems);
+        })
       } else {
         //
         // update existing grid layout, which is optimized to updates only diffs (will add new/delete items for examples)
         //
         const grid = gridRef.current;
-        const layout = [];
-        items.forEach((a) => layout.push(
+        const layout = items.map((a) => 
           refs.current[a.id].current.gridstackNode || {...a, el: refs.current[a.id].current}
```

Comment:
Regarding this line, i see that you are adding el property to each item, I also had to do so when i was writing a React wrapper component to avoid creating a new widget instead of updating. And i had to ts-ignore it, since i got typescript error, because the ` load()` function expects `Widget[]` as a parameter. It looks like a typing error in `load()`.

Reply:
yeah, the problem is the id YOU provide is not guaranteed to there nor unqiue, so the lib cannot depend on it being there... that's why I have _id internally to know what they are. so el is used (if present, else creates a new item).
I suppose I could check if id is set and assume callee does the right thing....

if you look at the Angular wrapper, I get the gridstackNode (which has _id and .el) to load, else it uses widget for new items positions...
https://github.com/gridstack/gridstack.js/blob/master/demo/angular/src/app/gridstack.component.ts#L129
with
https://github.com/gridstack/gridstack.js/blob/master/demo/angular/src/app/gridstack-item.component.ts#L39

Reply:
I shoudl document that better... but again have 2 list is the wrong approach, where as https://github.com/gridstack/gridstack.js/tree/master/demo/angular/src/app Angular wrapper is the right way.

Reply:
I have made my own react wrapper component over gridstack with a custom hook which can make any component draggable and GridStackDraggable component to make any react component gridstack draggable. Do you wish to have a look on it?

Reply:
that would be great - if you could post something similar to the Angular component I created, that would help others as I don't use React. Then peolple can interate and improved on it.